vendor_pricelist_category/models/product_supplierinfo.py

```diff
@@ -25,9 +24,10 @@ class ProductSupplierinfo(models.Model):
         "product_supplierinfo_category_id.discount",
     )
     def _compute_discount(self):
-        super()._compute_discount()
+        ret_val = super()._compute_discount()
         for sel in self.filtered(
```

Comment:
In this case, you also have to give values ​​to those that do not enter the filtered, otherwise, it gives strange errors.

Reply:
Those instances that do not meet the filter requirements have a value already, as the super function is called.

Reply:
ups, true 😄